src/main/java/org/broadinstitute/hellbender/tools/funcotator/mafOutput/MafOutputRenderer.java

```diff
@@ -0,0 +1,733 @@
+
+            // Get our funcotations for this allele and add them to the output maps:
+            for ( final Funcotation funcotation : funcotations ) {
+                if ( funcotation.getAltAllele().equals(altAllele) ) {
+                    // Add all the fields from the other funcotations into the extra field output:
+                    for ( final String field : funcotation.getFieldNames() ) {
+                        setField(extraFieldOutputMap, field, funcotation.getField(field));
+                    }
+                }
+            }
+
+            // Now add in our annotation overrides so they can be aliased correctly with the outputFieldNameMap:
+            extraFieldOutputMap.putAll(overrideAnnotations);
+
+            // Go through all output fields and see if any of the names in the value list are in our extraFieldOutputMap.
+            // For any that match, we remove them from our extraFieldOutputMap and add them to the outputMap with the
+            // correct key.
+            for ( final Map.Entry<String, List<String>> entry : outputFieldNameMap.entrySet() ) {
+                for ( final String fieldName : entry.getValue() ) {
+                    if ( extraFieldOutputMap.containsKey(fieldName) ) {
+                        outputMap.put(entry.getKey(), extraFieldOutputMap.remove(fieldName));
+                        break;
+                    }
+                }
+            }
+
+            // Merge our output maps together:
+            outputMap.putAll(extraFieldOutputMap);
+
+            // Now translate fields to the field names that MAF likes:
+            final LinkedHashMap<String, String> mafCompliantOutputMap = replaceFuncotationValuesWithMafCompliantValues(outputMap);
+
+            // Write our header if we have to:
+            if ( ! hasWrittenHeader ) {
+                writeHeader(mafCompliantOutputMap);
+            }
+
+            // Write the output (with manual annotations at the end):
+            writeLine(
+                    mafCompliantOutputMap.values().stream().collect(Collectors.joining(MafOutputRendererConstants.FIELD_DELIMITER))
+                    + manualAnnotationSerializedString
+            );
+        }
+    }
+
+
+    //==================================================================================================================
+    // Static Methods:
+
+    //==================================================================================================================
+    // Instance Methods:
+
+    /**
+     * Ensures each value in the given {@code outputMap} is compliant with MAF-expected values.
+     * @param outputMap The {@link Map} of output field -> values to be checked for MAF compliance.
+     * @return A {@link LinkedHashMap} of output field strings -> values to be written to the MAF file.
+     */
+    private LinkedHashMap<String, String> replaceFuncotationValuesWithMafCompliantValues(final Map<String, Object> outputMap ) {
```

Comment:
I think there should be unit tests for this, and for the `write`.  That probably means extracting a method from within write that returns the output string , or the array of things to write instead actually writing to disk.  I think at a minimum we need a test to show that we're correctly handling  deletion and insertion alleles.  As well as basic conversion, defaulting, and overriding is working.  We don't need exhaustive tests for every case right now, but there should be something.

Reply:
Fair point.  I'll add the tests when I build out the `MafOutputRendererUnitTest`.